app/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -2485,6 +2485,10 @@ private void setSortListGroup(boolean currentListGroupVisibility, boolean show)
      * visibility earlier using {@link #setSortListGroup(boolean, boolean)}
      */
     private void popSortListGroupVisibility() {
+        if (previousSortGroupState.isEmpty()) {
+            return;
+        }
+
         boolean popped = previousSortGroupState.pop();
         showSortListGroup(popped);
```

Comment:
```suggestion
        if (!previousSortGroupState.isEmpty()) {
            boolean popped = previousSortGroupState.pop();
            showSortListGroup(popped);
        }
```

In my opinion this version is a touch cleaner.

Reply:
IMO avoid [nesting code](https://stackoverflow.com/a/268187/16787943) better.